src/editor/shadow-taxonomies/index.js

```diff
@@ -0,0 +1,274 @@
+								</p>
+							) }
+							checked={ newspack_listings_hide_parents }
+							onChange={ value => updateMetaValue( 'newspack_listings_hide_parents', value ) }
+						/>
+					</PanelRow>
+
+					{ ! newspack_listings_hide_parents &&
+						Object.keys( taxonomies ).map( slug => {
+							const { name, post_types: childPostTypesForTaxonomy } = taxonomies[ slug ];
+							const taxonomyPostType = getPostTypeByTaxonomy( name );
+
+							/**
+							 * Only show shadow terms for listing types that can own this post type.
+							 * Don't show shadow terms for the same type of listing as the current post.
+							 */
+							if (
+								-1 === childPostTypesForTaxonomy.indexOf( postType ) ||
+								taxonomyPostType === postType
+							) {
+								return null;
+							}
+
+							const terms = getEditedPostAttribute( name );
+							return (
+								<TaxonomySearch
+									key={ slug }
+									postId={ postId }
+									postTitle={ getEditedPostAttribute( 'title' ) }
+									savedIds={ terms }
+									fetchSaved={ async ( ids, tax ) => {
+										return await apiFetch( {
+											path: addQueryArgs( '/newspack-listings/v1/terms', {
+												per_page: 100,
+												_fields: 'id,name',
+												taxonomy: tax,
+												include: ids.join( ',' ),
+											} ),
+										} );
+									} }
+									fetchSuggestions={ async ( search, tax ) => {
+										return await apiFetch( {
+											path: addQueryArgs( '/newspack-listings/v1/terms', {
+												search,
+												per_page: 20,
+												_fields: 'id,name',
+												orderby: 'count',
+												order: 'desc',
+												taxonomy: tax,
+											} ),
+										} );
+									} }
+									taxonomy={ taxonomies[ slug ] }
+									update={ termIds => {
+										const newTerms = {};
+										newTerms[ name ] = termIds;
+										editPost( newTerms );
+									} }
+								/>
+							);
+						} ) }
+				</PluginDocumentSettingPanel>
+			) }
+			{ canHaveChildren && (
+				<PluginDocumentSettingPanel
+					className="newspack-listings__shadow-taxonomies-sidebar"
+					name="newspack-listings-children"
+					title={ __( 'Child Listings', 'newspack-listings' ) }
+				>
+					<PanelRow>
+						<ToggleControl
+							className={ 'newspack-listings__toggle-control' }
+							label={ __( 'Hide child listings', 'newspack-listings' ) }
+							help={ () => (
+								<p>
+									{ __( 'Overrides ', 'newspack-listings' ) }
+									<ExternalLink href="/wp-admin/admin.php?page=newspack-listings-settings-admin">
+										{ __( 'global settings', 'newspack-listings' ) }
+									</ExternalLink>
+								</p>
+							) }
+							checked={ newspack_listings_hide_children }
+							onChange={ value => updateMetaValue( 'newspack_listings_hide_children', value ) }
+						/>
+					</PanelRow>
+
+					{ ! newspack_listings_hide_children &&
+						filteredChildPostTypes.map( childPostType => {
+							return (
+								<TaxonomySearch
```

Comment:
When editing a marketplace, I see posts & pages field. After inputing text, nothing is found and the request returns `404`. 

![image](https://user-images.githubusercontent.com/7383192/115691983-d8885a00-a35e-11eb-96c8-7c7c802f2502.png)

Reply:
Fixed in 527fe4d—forgot that the core `posts` and `pages` endpoint have the plural `s`.

Also in the same commit: since only published listings get shadow terms, we now hide the Child Listings UI if the post you're editing isn't published. Also added a `validate_post` method to the taxonomies class to make it a bit easier to check whether a post should have a shadow term.

Reply:
The request now works, but I see an error in the logs: 
```
Trying to get property 'term_id' of non-object
```
Originating from here: 

https://github.com/Automattic/newspack-listings/blob/527fe4d35e4b1df818a4c41a8e10d656c8afa8bf/includes/class-newspack-listings-taxonomies.php#L578

Where `$shadow_term` is false, when I'm linking a just-published Event to a post or a page.

Reply:
I made some logical errors in the `validate_post` method in the prior commit. Should be fixed in 4b43a7c.